e2e/testcases/multi_sync_test.go

```diff
@@ -407,10 +406,14 @@ func TestConflictingDefinitions_RootToRoot(t *testing.T) {
 	nt.RootRepos[rootSync2].Add(podRoleFilePath, rootPodRole())
 	nt.RootRepos[rootSync2].CommitAndPush("add conflicting pod owner role")
 
-	nt.T.Logf("The admission webhook should deny the update request in Root %s", rootSync2)
-	nt.WaitForRootSyncSyncError(rootSync2, applier.ApplierErrorCode, "denied the request")
-	nt.T.Logf("Root %s should also get surfaced with the conflict error", configsync.RootSyncName)
-	nt.WaitForRootSyncSyncError(configsync.RootSyncName, status.ManagementConflictErrorCode, "declared in another repository")
+	// When the webhook is enabled, it will stop adoption of managed objects.
+	// So only the second RootSync will show the conflict error.
```

Comment:
The second RootSync will show the conflict error because the admission webhook denied the update. Then, the remediator of the second RootSync will detect the drift and surface the conflict to both RootSyncs. The reason for reporting to both is that the object can be removed from either of the repos.

Here is the logic to report the conflict error for the second RootSync:
https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/run.go#L422.

Here is how to surface a similar error to the conflicting RootSync: https://github.com/GoogleContainerTools/kpt-config-sync/blob/main/pkg/parse/run.go#L427.

Reply:
I see. So by removing `UpdateConflictManagerStatus` and just merging `p.RemediatorConflictErrors()`  I broke the "logic to report the conflict error for the second RootSync".

It wasn't super clear what that logic was doing. But I see now it's calling `conflictError.ConflictingManagerError()` on every element of the `[]ManagementConflictError`, which I guess inverts the error.

Why isn't that logic all just in the Remediator? It seems pretty convoluted to need to process the ManagementConflictErrors into a MultiError in the parser.

Reply:
I've reverted this change and instead migrated both `WaitForRootSyncSyncError` calls to use WaitForObject in parallel instead. This should evaluate faster and more consistently, without actually changing the existing behavior, except to make it less flakey.